### PR TITLE
Updates dependencies

### PR DESCRIPTION
**Pull request by Meronat**
This goes ahead and updates a lot of the dependencies. I was originally going to just update Guava and GSON for Sponge compatibility in 1.12 as Guava introduced a breaking change, but I thought updating other dependencies wouldn't hurt. 

I did have one issue though that's why this is WIP, as I cannot get the following HOCON test to pass, so Configurate will not compile(So this test seems to only not pass locally on my Windows machine):

image

It appears the output does mess up the comment. Here is the surefire report:
https://gist.github.com/Meronat/5a5e3722e7b9636aa2b4e39b7bc57f1c

### CODE REVIEW

**Comment by Meronat** [comment]
Woo finally got it to compile. The mentioned test still fails locally though if anyone can provide some insight into fixing that.

**Comment by zml2008** [comment]
I think that test was something that failed on windows but not on Linux -- I'm not sure if I've been able to track it down yet either.

File: configurate-core/pom.xml [hunk 1]
```diff
@@ -36,7 +36,7 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <version>17.0</version>
+            <version>21.0</version>
```
Comment (zml2008) [inline configurate-core/pom.xml]: We can't update Guava until MC/Sponge updates its Guava version

File: configurate-core/pom.xml [hunk 2]
```diff
@@ -21,7 +21,7 @@
   <parent>
       <groupId>ninja.leaping.configurate</groupId>
       <artifactId>configurate-parent</artifactId>
-      <version>3.2.1-SNAPSHOT</version>
+      <version>3.3</version>
```
Comment (zml2008) [inline configurate-core/pom.xml]: The only time the version field should be changed is by me when I'm making a release with maven release plugin, not you.

File: configurate-gson/pom.xml [hunk 3]
```diff
@@ -34,7 +34,7 @@
         <dependency>
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
-            <version>2.2.4</version>
```
Comment (zml2008) [inline configurate-gson/pom.xml]: Again, we can't change this version since MC/Sponge use gson 2.2.4

Comment (kinggoesgaming) [inline configurate-gson/pom.xml]: although keep in mind that 1.12 is updating to gson 2.8.0

**Comment by Meronat** [comment]
@zml2008 I'll go ahead and change the version back. What should I set the version as in the changelog though?

And yeah the point of the PR is to update GSON and Guava to the versions Sponge will be using with the Minecraft 1.12 release coming soon as that updates both dependencies. For the versions they don't need these changes, they can continue using the same version and for 1.12 branches they can up the Configurate dependency :)

**Comment by kashike** [comment]
There's no reason configurate can't update dependencies now. Sponge doesn't need to update configurate just because a new version is released.

**Comment by Meronat** [comment]
This should be good to go. Any other changes you'd like to see? 

1.12 hit pre-release this week, so I doubt we'll see any library changes.

**Comment by kashike** [comment]
It'd be great to get this merged sometime soon.

**Pull request closed without merging**